index.js

```diff
@@ -91,47 +91,65 @@ OpenDKIMVerifyStream.prototype._build_result = function(error) {
   return result;
 };
 
-OpenDKIMVerifyStream.prototype.write = function(buf) {
-  if (buf && buf.length && !(this.finished)) {
-    var chunk = this._get_chunk(buf);
-    try {
-      this.opendkim.chunk({
+OpenDKIMVerifyStream.prototype.write = function (buf) {
+  var self = this;
+
+  if (buf && buf.length && !(self.finished)) {
+    var chunk = self._get_chunk(buf);
+    var options = {
         message: chunk,
         length: chunk.length
-      });
-    } catch (err) {
-      return this.callback(err, this._build_result(err));
-    }
-  }
+    };
+    self.opendkim.chunk(options)
+    .then(result => {
+      self.emit('drain');
+    }).catch(error => {
+      self.callback(error, self._build_result(error));
+    });
 
-  return true;
+    return false;
+  } else {
+    return true;
+  }
```

Comment:
I always prefer short clauses like this to get nestled up with the conditional, maybe something like:

```js
if (!buf || !buf.length || self.finished) {
  return true;
}
```

And then the rest of the method is saved one tab level, too.